app/src/main/java/fr/free/nrw/commons/explore/map/ExploreMapFragment.java

```diff
@@ -156,43 +157,34 @@ public class ExploreMapFragment extends CommonsDaggerSupportFragment
     @BindView(R.id.category)
     TextView distance;
 
-    private ActivityResultLauncher<String[]> activityResultLauncher = registerForActivityResult(
-        new ActivityResultContracts.RequestMultiplePermissions(),
-        new ActivityResultCallback<Map<String, Boolean>>() {
-            @Override
-            public void onActivityResult(Map<String, Boolean> result) {
-                boolean areAllGranted = true;
-                for (final boolean b : result.values()) {
-                    areAllGranted = areAllGranted && b;
-                }
-
-                if (areAllGranted) {
-                    locationPermissionGranted();
+    private ActivityResultLauncher<String> activityResultLauncher = registerForActivityResult(
```

Comment:
Ditto here about replacing this with an invocation to `LocationPermissionHelper`'s `requestForLocationAccess` .